src/components/Chat/BottomSheetGif.js

```diff
@@ -48,15 +41,34 @@ const BottomSheetGif = React.forwardRef((props, ref) => {
     setIsLoading(false);
   };
 
+  const searchGIF = React.useCallback(
+    debounce((search) => {
+      onLoad(search);
+    }, 500),
+    []
+  );
+
+  const handleChangeText = (text) => {
+    setSearchText(text);
+    if (text?.trim()?.length > 2) {
+      searchGIF(text);
+    }
+  };
+
+  const handleOnClearText = () => {
+    setSearchText('');
+    searchGIF('');
+  };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `searchGIF` function is debounced, which is a good practice for search input handling as it prevents unnecessary API calls. However, there's no check for an empty string before calling `onLoad(search)`. If the user clears the search input, this will trigger an unnecessary API call with an empty query. Consider adding a condition to prevent this.
```diff
  const searchGIF = React.useCallback(
    debounce((search) => {
+     if (search.trim() !== '') {
        onLoad(search);
+     }
    }, 500),
    []
  );
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->